Breadcrumbs.py

```diff
@@ -108,4 +141,223 @@ def get_points_by_row(row):
               breadcrumbs[index_of_breadcrumb] = breadcrumbs[index_of_breadcrumb][0:length_of_trim]
           break
       number_of_characters_left -= current_length
-    view.set_status('breadcrumbs',separator.join(breadcrumbs+['']))
+  return breadcrumbs
+
+
+def copy(view, text):
+  sublime.set_clipboard(text)
+  sublime.status_message('Breadcrumbs copied to clipboard')
+
+
+if viewevents_available is not True:
+  class BreadcrumbsEventListenerST2(sublime_plugin.EventListener):
+
+    def on_selection_modified(self, view):
+      if get_statusbar_enabled(view.settings()):
+        current_row = view.rowcol(view.sel()[0].b)[0]
+        breadcrumbs = make_breadcrumbs(view, current_row, shorten=True)
+
+        if len(breadcrumbs) > 0:
+          view.set_status('breadcrumbs', get_separator(view).join(breadcrumbs))
+      else:
+        view.erase_status('breadcrumbs')
+
+
+class BreadcrumbsEventListenerST3(sublime_plugin.ViewEventListener):
+
+  @classmethod
+  def is_applicable(cls, settings):
+    return get_statusbar_enabled(settings) and viewevents_available
```

Comment:
This `and viewevents_available` can be dropped once you wrap the whole thing in if/else, which you have to do anyway.